libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -431,5 +442,10 @@ private bool ChannelSupportsOAuthCard(string channelId)
 
             return true;
         }
+
+        private bool RequiresLink(ITurnContext turnContext)
+        {
+            return !turnContext.Activity.ServiceUrl.Contains("http");
```

Comment:
I don't understand this. Can you add in a detailed comment explaining the logic here? If nothing else, I don't understand why we would ever allow this over http vs https.

Reply:
+1, is the logic here that if we are using websockets or pipes this will return true, and for http / https return false? If that is the case, this logic should be documented. Documentation could even start at the class design level, i.e. you could have a small abstraction that given an activity, determines whether it is from Pipes, Sockets, or Http. But yes, at least add comments. What are the possible values? what are the scenarios behind each value? why do some scenarios require  a link and others not? future maintainers of this code need to understand this decision fully in the future.

Reply:
This is documented in OBI, which is not public, so I will add a comment.